tests/unittests/sources/test_vmware.py

```diff
@@ -463,336 +462,312 @@ def test_extra_hotplug_udev_rules(self):
         )
 
 
-class TestDataSourceVMwareEnvVars(FilesystemMockingTestCase):
+class TestDataSourceVMwareEnvVars:
     """
     Test the envvar transport.
     """
 
-    def setUp(self):
-        super(TestDataSourceVMwareEnvVars, self).setUp()
-        self.tmp = self.tmp_dir()
-        os.environ[DataSourceVMware.VMX_GUESTINFO] = "1"
-        self.create_system_files()
-
-    def tearDown(self):
-        del os.environ[DataSourceVMware.VMX_GUESTINFO]
-        return super().tearDown()
-
-    def create_system_files(self):
-        rootd = self.tmp_dir()
+    @pytest.fixture(autouse=True)
+    def env_and_files(self, fake_filesystem, monkeypatch, tmpdir):
+        monkeypatch.setenv(DataSourceVMware.VMX_GUESTINFO, "1")
         populate_dir(
-            rootd,
-            {
-                DataSourceVMware.PRODUCT_UUID_FILE_PATH: PRODUCT_UUID,
-            },
+            str(tmpdir),
```

Comment:
Theoretically, we can pass an empty string instead of `str(tmpdir)` as the `fake_filesystem` is already handling prepending the path inside the `os` module calls of `populate_dir`. However, in my opinion, an explicit `str(tmpdir)`  helps in term of code clarity and future contributors.

Up to the reviewer to decide which way I should go